torch/_inductor/compile_fx.py

```diff
@@ -543,17 +543,23 @@ def maybe_realign_inputs(
             compiled_graph.current_callable = new_callable
 
 
+def compile_fx_inner(*args, **kwargs):
+    with torch.utils._python_dispatch._disable_current_modes(), _use_lazy_graph_module(
+        dynamo_config.use_lazy_graph_module
+    ), dynamo_utils.dynamo_timed(
+        "compile_fx_inner", phase_name="inductor_compile", fwd_only=False
+    ):
+        return _compile_fx_inner(*args, **kwargs)
+
+
 @DebugContext.wrap
-@torch.utils._python_dispatch._disable_current_modes()
 @time_and_log(attr="compilation time (in seconds)")
 # Need this decorator for compile_fx_inner even if we already have one for
 # compile_fx. The reason is the compilation for backward graph may happen after
 # compile_fx return and we may want to use the _LazyGraphModule for compiling
 # the backward graph as well.
-@_use_lazy_graph_module(dynamo_config.use_lazy_graph_module)
 @with_fresh_cache_if_config
-@dynamo_utils.dynamo_timed(phase_name="inductor_compile", fwd_only=False)
-def compile_fx_inner(
+def _compile_fx_inner(
```

Comment:
These decorators are temporarily reordered, but I don't think it matters. Once all of these become context managers I can restore original order.